cmd/sonobuoy/app/operations/cp.go

```diff
@@ -17,10 +17,148 @@ limitations under the License.
 package operations
 
 import (
+	"archive/tar"
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	podName       = "sonobuoy"
+	podNamespace  = "heptio-sonobuoy"
+	containerName = "kube-sonobuoy"
+	results       = "/tmp/sonobuoy"
 )
 
-func CopyResults(path string) error {
-	// pretty straight forwards
-	return errors.New("not implemented")
+// CopyResults copies results from a sonobuoy run into a Reader in tar format.
+func CopyResults(config *rest.Config, cmderr io.Writer, errc chan error) io.Reader {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		errc <- fmt.Errorf("unable to create clientset: %v", err)
+		return nil
+	}
+	client := clientset.CoreV1().RESTClient()
+	req := client.Post().
+		Resource("pods").
+		Name(podName).
+		Namespace(podNamespace).
+		SubResource("exec").
+		Param("container", containerName)
+	req.VersionedParams(&corev1.PodExecOptions{
+		Container: containerName,
+		Command:   []string{"tar", "cf", "-", results},
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+	}, scheme.ParameterCodec)
+	executor, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		errc <- fmt.Errorf("unable to get remote executor: %v", err)
+		return nil
+	}
+	reader, writer := io.Pipe()
+	go func() {
```

Comment:
Why not make this blocking?

Reply:
so it came like this and i thought about why not make this blocking:

If the file is large, we can stream it straight to disk with this method and not buffer the entire thing in memory. If we make this blocking we'll have to buffer the entire tarball into memory then write to disk.

If you think the complexity of this code outweighs the likelihood of us caring about the size of the tarball I'll be happy to write it blocking, much code can be removed.